packages/flutter_test/lib/src/accessibility.dart

```diff
@@ -196,131 +234,173 @@ class MinimumTextContrastGuideline extends AccessibilityGuideline {
-      // If rectangle is empty, pass the test.
-      if (report.isEmptyRect) {
-        return result;
+  Future<Evaluation> _evaluateNode(
+    SemanticsNode node,
+    WidgetTester tester,
+    ui.Image image,
+    ByteData byteData,
+  ) async {
+    Evaluation result = const Evaluation.pass();
+
+    // Skip disabled nodes, as they not required to pass contrast check.
+    final bool isDisabled = node.hasFlag(ui.SemanticsFlag.hasEnabledState) &&
+        !node.hasFlag(ui.SemanticsFlag.isEnabled);
+
+    if (node.isInvisible ||
+        node.isMergedIntoParent ||
+        node.hasFlag(ui.SemanticsFlag.isHidden) ||
+        isDisabled) {
+      return result;
+    }
+
+    final SemanticsData data = node.getSemanticsData();
+    final List<SemanticsNode> children = <SemanticsNode>[];
+    node.visitChildren((SemanticsNode child) {
+      children.add(child);
+      return true;
+    });
+    for (final SemanticsNode child in children) {
+      result += await _evaluateNode(child, tester, image, byteData);
+    }
+    if (shouldSkipNode(data)) {
+      return result;
+    }
+
+    // Look up inherited text properties to determine text size/weight/color.
+    late bool isBold;
+    double? fontSize;
+    Color? textColor;
+
+    final String text = data.label.isEmpty ? data.value : data.label;
+    final List<Element> elements = find.text(text).hitTestable().evaluate().toList();
+    late final Rect paintBounds;
+
+    if (elements.length == 1) {
+      final Element element = elements.single;
+      final RenderObject? renderBox = element.renderObject;
+      if (renderBox is! RenderBox) {
+        throw StateError('Unexpected renderObject type: $renderBox');
       }
-      final double contrastRatio = report.contrastRatio();
-      const double delta = -0.01;
-      double targetContrastRatio;
-      if ((isBold && (fontSize ?? _kDefaultFontSize) > kBoldTextMinimumSize) || (fontSize ?? _kDefaultFontSize) > kLargeTextMinimumSize) {
-        targetContrastRatio = kMinimumRatioLargeText;
+
+      const Offset offset = Offset(4.0, 4.0);
+      paintBounds = Rect.fromPoints(
+        renderBox.localToGlobal(renderBox.paintBounds.topLeft - offset),
+        renderBox.localToGlobal(renderBox.paintBounds.bottomRight + offset),
+      );
+      final Widget widget = element.widget;
+      final DefaultTextStyle defaultTextStyle = DefaultTextStyle.of(element);
+      if (widget is Text) {
+        final TextStyle? style = widget.style;
+        final TextStyle effectiveTextStyle = style == null || style.inherit
+            ? defaultTextStyle.style.merge(widget.style)
+            : style;
+        isBold = effectiveTextStyle.fontWeight == FontWeight.bold;
+        fontSize = effectiveTextStyle.fontSize;
+        textColor = effectiveTextStyle.color;
+      } else if (widget is EditableText) {
+        isBold = widget.style.fontWeight == FontWeight.bold;
+        fontSize = widget.style.fontSize;
+        textColor = widget.style.color;
```

Comment:
This will not work.

In addition to specifying the `color`, a user might also or instead specify a foreground paint. That paint may have a null `color` and instead set a `shader` to draw some kind of gradient on the text. That shader is completely opaque to this code. Alternatively, a user might intentionally add a `maskFilter` to blur the text, or ` colorFilter` to change the specified color, both of which are also opaque to this code.

Alternatively, the text may be composed into a layer that will change its visual appearance/color, such as a `ColorFilterLayer` or an `ImageFilterLayer` or an `OpacityLayer`. 

Alternatively, some text implementation detail may alter or transform the color as specified by the widget in some non-obvious way. This code will break without any insight into that change.

Reply:
Hi,
I have discussed with jonahwilliams regarding font color "sniffing" I will remove that for this PR, as his had similar concerns. 

Will file issue on github, as test/heuristic in current state is broken for small fonts.

Reply:
Removed text-color. 
Keeping other fixes.